### PR TITLE
convert path params with colon delimiter to bracelet delimiter during spec generation

### PR DESCRIPTION
**Pull request by itamarco**
### All Submissions:

* [x] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
* [x] Have you written unit tests?
* [x] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
* [x] This PR is associated with an existing issue?

**Closing issues**

Put `closes #XXXX`  (where XXXX is the issue number) in your comment to auto-close the issue that your PR fixes.

### If this is a new feature submission:

* [x] Has the issue had a maintainer respond to the issue and clarify that the feature is something that aligns with the [goals](https://github.com/lukeautry/tsoa#goal) and [philosophy](https://github.com/lukeautry/tsoa#philosophy) of the project?

**Potential Problems With The Approach**

**Test plan**

* I have wrote unit tests for the `convertColonPathParams` method in `pathUtils.ts`
* I have added route with colon delimiter path params to `paramaeterController.ts` and tested the resulted spec

### CODE REVIEW

File: src/utils/pathUtils.ts [hunk 1]
```diff
@@ -18,3 +19,12 @@ export function normalisePath(path: string, withPrefix?: string, withSuffix?: st
   normalised = normalised.replace(/[/\\\s]+/g, '/');
   return normalised;
 }
+
+export function convertColonPathParams(path: string) {
```
Comment (WoH) [inline src/utils/pathUtils.ts]: minor nitpick: I think it's reasonable to do this inside `normalizePath`

Comment (itamarco) [reply src/utils/pathUtils.ts]: I agree and I put it there at first, but it made route generation tests fail.

File: src/utils/pathUtils.ts [hunk 2]
```diff
@@ -18,3 +19,12 @@ export function normalisePath(path: string, withPrefix?: string, withSuffix?: st
...
+  if (!path || typeof path !== 'string') {
+    return path;
+  }
+
+  const normalised = path.replace(/:([^/]+)/g, '{$1}');
```
Comment (WoH) [inline src/utils/pathUtils.ts]:
```suggestion
  const normalised = path.replace(/\/:([^/]+)/g, '{$1}');
```

File: src/utils/pathUtils.ts (hunk 1 above)
Comment (WoH) [reply src/utils/pathUtils.ts]: Yes, we are calling this again while building the templates. I assume this is the better approach then.

File: src/utils/pathUtils.ts (hunk 2 above)
Comment (WoH) [inline src/utils/pathUtils.ts]:
```suggestion
  const normalised = path.replace(/:([^\/]+)/g, '{$1}');
```

**Pull request merged**